### PR TITLE
Add max measurements to Micrometer Timer & DistributionSummary

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Fixes one of the issues discovered in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5292

### CODE REVIEW

**Review by trask — APPROVED** [review]

File: instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java [hunk 1]
```diff
@@ -197,4 +198,49 @@ void testMicrometerPercentiles() {
...
+                            .anySatisfy(
+                                point ->
+                                    assertThat(point)
+                                        .hasValue(4_000)
+                                        .attributes()
+                                        .containsEntry("tag", "value"))));
+
+    // when
+    Metrics.globalRegistry.remove(timer);
+    Thread.sleep(10); // give time for any inflight metric export to be received
```
Comment (anuraaga) [inline instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java:237]: Will be able to remove this after #5314 I think

**Review by anuraaga — APPROVED** [review]

File: instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java (hunk 1 above)
Comment (trask) [reply instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java:237]: I'm guessing there may still be a tiny window while the metric sdk is still collecting metrics but before it has sent them to the metric exporter(?)

Comment (anuraaga) [reply instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java:237]: At time of writing, that PR was causing clearData to first flush the metrics SDK. But realized it's not necessary or helpful in most tests so changed it back, so indeed this comment is now invalid

Comment (trask) [reply instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java:237]: oh i c, it flushed the full collection cycle, not just the export? that does seem useful for all of these micrometer unregister tests that we have

**Pull request merged**